webcompat/api/uploads.py

```diff
@@ -53,8 +53,8 @@ def to_image_object(self, imagedata):
                 # Chop off 'data:image/.+;base64,' before decoding
                 imagedata = re.sub('^data:image/.+;base64,', '', imagedata)
                 return Image.open(BytesIO(base64.b64decode(imagedata)))
-            raise IOError('Not a valid image format')
-        except IOError as e:
+            raise TypeError('TypeError: Not a valid image format')
```

Comment:
Yes. Sure. Let's read the documentation for errors

* [IOError](https://docs.python.org/2/library/exceptions.html#exceptions.IOError)  
> Raised when an I/O operation (such as a print statement, the built-in open() function or a method of a file object) fails for an I/O-related reason, e.g., “file not found” or “disk full”.
* [TypeError](https://docs.python.org/2/library/exceptions.html#exceptions.TypeError)  
>  Raised when an operation or function is applied to an object of inappropriate type. The associated value is a string giving details about the type mismatch.

My feeling is that semantically we are bailing out because our code didn't get the appropriate format of image, more than the issue with the file being existent or not on the disk.